### ISSUE TITLE
Make database collation configurable

### ISSUE THREAD

**Issue by frostieDE** [sameersbn/docker-gitlab#1178]
Note: this issue is MySQL-specific.

To make comments work with emojis, the database collation must be `utf8mb4_general_ci` instead of `utf8_general_ci` which is hardcoded inside [database-yml](https://github.com/sameersbn/docker-gitlab/blob/master/assets/runtime/config/gitlabhq/database.yml) 

But it seems that hardcoding it to `utf8mb4_general_ci` isn't a good solution as well, which leads to my wish of making this parameter configurable :-)

Also see https://gitlab.com/gitlab-org/gitlab-ce/issues/30698#note_27228995

**Comment by frostieDE** [comment]
Just saw that there is already an opened pull request #1098

**Comment by phpb-com** [comment]
Will this help? https://docs.gitlab.com/ce/administration/environment_variables.html

**Comment by frostieDE** [comment]
No it does not seem to solve the problem. But they do not state that I can change the `collation` setting (just `encoding` which I can already set). I tried setting the following variables:

```
-DB_ENCODING=utf8mb4
- GITLAB_DATABASE_COLLATION=utf8mb4_general_ci
```
This makes GitLab not respond to any request as it has problems speaking with MySQL.

**Comment by phpb-com** [comment]
I see. It is definitely need fixing as per https://gitlab.com/gitlab-org/gitlab-ce/blob/master/doc/install/database_mysql.md

**Comment by thiago** [comment]
Any updates about this?

**Comment by solidnerd** [comment]
Hey, 
there is an PR for this #1098 . Could someone review it ?

**Comment by solidnerd** [comment]
It was Implemented by #1262  and is available in `9.3.0`.  I will close this issue and the the problems   still exists please reopen the issue and ping someone.
